cluster/manifests/z-karpenter/deployment.yaml

```diff
@@ -0,0 +1,128 @@
+{{ if karpenterNodePools .Cluster.NodePools }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: karpenter
+  namespace: kube-system
+  labels:
+    application: kubernetes
+    component: karpenter
+spec:
+  replicas: 2
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      deployment: karpenter
+  template:
+    metadata:
+      labels:
+        deployment: karpenter
+        application: kubernetes
+        component: karpenter
+      annotations:
+        logging/destination: "{{.Cluster.ConfigItems.log_destination_infra}}"
+    spec:
+      dnsConfig:
+        options:
+          - name: ndots
+            value: "1"
+      serviceAccountName: karpenter
+      securityContext:
+        fsGroup: 1000
+      priorityClassName: "{{ .Cluster.ConfigItems.system_priority_class }}"
+      tolerations:
+      - key: node.kubernetes.io/role
+        value: master
+        effect: NoSchedule
+      # TODO: should or can run on master?
+      # nodeSelector:
+      #   node.kubernetes.io/role: master
```

Comment:
I think it makes sense to run it on master nodes (It allows for backing all worker nodes by karpenter). Note, that we run cluster-autoscaler as a DaemonSet instead of a Deployment (also on the masters).